pilot/pkg/model/network_test.go

```diff
@@ -89,9 +90,35 @@ func TestGatewayHostnames(t *testing.T) {
 		// after the update, we should see the next gateway. Since TTL is low we don't know the exact IP, but we know it should change from
 		// the original
 		retry.UntilOrFail(t, func() bool {
-			return !reflect.DeepEqual(env.NetworkManager.AllGateways(), []model.NetworkGateway{{Network: "nw0", Addr: firstIP, Port: 15443}})
+			return !reflect.DeepEqual(env.NetworkManager.AllGateways(), gateways)
 		})
+		xdsUpdater.WaitOrFail(t, "xds full")
+	})
+
+	workingDNSServer.setHosts(make(sets.Set[string]))
+	gateways = env.NetworkManager.AllGateways()
+	t.Run("resolution failed", func(t *testing.T) {
+		// the gateways should remain
+		retry.Until(func() bool {
```

Comment:
retry function should not call Fatal in the function, should use `retryOrFail` if you want to fail.

Right now this ALWAYS runs for 1s which is unexpected. Same with L111

Reply:
Or if we mean to check "remains the same for 1s" we can use other mechanisms, like just `xdsUpdater.AssertEmpty` and then check the AllGateways once. And use a lower time than 1s like 50ms

Reply:
Thank you, fixed